src/containers/ConnectorManagement.jsx

```diff
@@ -171,10 +174,18 @@ export default class ConnectorManagement extends Component {
           .fetchAccounts(this.props.params.connectorSlug, null)
           .then(accounts => {
             konnector.accounts = accounts
-            this.setState({
-              connector: konnector,
-              isConnected: konnector.accounts.length !== 0,
-              submitting: false
+            const currentIdx = accounts.findIndex(a => a._id === accountID)
```

Comment:
Reminder for everyone : we should really stop using array index for getting the current account, it is so an error prone solution.